### ISSUE TITLE
ARM64 architecture in Debian 9 repository

### ISSUE THREAD

**Issue by Petermarcu** [dotnet/runtime#3298]
@Levi--G commented on [Sun Aug 05 2018](https://github.com/dotnet/core/issues/1840)

# ARM64 architecture in Debian repository

I just tried installing ARM64 runtime with the package manager and apparently its not in there.

# Details

This is more of a question/documentation clarification than an actual issue, but it was annoying enough for me to make this.
So after going to [https://www.microsoft.com/net/download/dotnet-core/2.1](https://www.microsoft.com/net/download/dotnet-core/2.1) I got the impression I could also install arm64 runtime through the package manager since there are no limitations listed.

![](https://gyazo.com/2270b0e96a39364619bdc1e222a3cace.png)

But unfortunately after following the guide at [https://www.microsoft.com/net/download/linux-package-manager/debian9/runtime-2.1.2](https://www.microsoft.com/net/download/linux-package-manager/debian9/runtime-2.1.2) (still no limitations listed to amd64 only) I found out the repo doesn't contain an arm package, so I had to manually download the tar etc... Now that isn't that bad but I would rather have known that sooner, is it possible to mention this in the listing of the versions? Is there a package planned for arm? I would expect it to not be that much effort (but then again I am only a noob what do I know 🤣)

Friendly regards,

---

@leecow commented on [Mon Aug 06 2018](https://github.com/dotnet/core/issues/1840#issuecomment-410938359)

Excellent points. Thanks @Levi--G . There are currently not plans to add package manger support for ARM64 but that could certainly change. Will work on some dl instruction content to make it clear that it's all referring to AMD64.

cc @vivmishra 

---

@vielmetti commented on [Mon Aug 13 2018](https://github.com/dotnet/core/issues/1840#issuecomment-412659820)

Thanks @Levi--G @leecow . I too would like to see arm64 support in Debian packaging for the runtime.

---

@Petermarcu commented on [Thu Sep 06 2018](https://github.com/dotnet/core/issues/1840#issuecomment-419239714)

@dleeapho as FYI

**Comment by dleeapho** [comment]
Let's take do this in 6.0 when we have time to 'close with' any package authoring toolset open questions.

**Comment by EdDore** [comment]
How about stepping back and doing this for .NET 6.0, which is an LTS release. I can't believe there is no ARM64 package install for this LTS release. There is certainly something to be said for install consistency across the different linux architectures here. Not having a simple package manager install for this, makes installing this quite error prone for us linux challenged users, coming from a windows background.

**Comment by danmoseley** [comment]
Cc @richlander

**Comment by richlander** [comment]
There is a project to do this and we are in favor it. I believe it's stuck behind some other projects in our backlog.

Is that right @MichaelSimons @leecow?

**Comment by MichaelSimons** [comment]
Yes it's on our backlog and didn't make the 7.0 cut because of other higher priority work.

**Comment by AnalogFeelings** [comment]
.NET 9 has launched, and this still hasn't been solved. Is it ever going to be?

**Comment by Aldaviva** [comment]
@AnalogFeelings I'm not sure if this matches your requirements or is something you'd want to use, but I just updated my unofficial [ARM32/ARM64 APT repository](https://github.com/Aldaviva/RaspberryPiDotnetRepository) with new DEB packages for this week's .NET release, including .NET 9.0.0 GA packages.

And I agree, it would be great if this were ever officially solved upstream instead.

**Comment by AnalogFeelings** [comment]
> @AnalogFeelings I'm not sure if this matches your requirements or is something you'd want to use, but I just updated my unofficial [ARM32/ARM64 APT repository](https://github.com/Aldaviva/RaspberryPiDotnetRepository) with new DEB packages for this week's .NET release, including .NET 9.0.0 GA packages.
> 
> And I agree, it would be great if this were ever officially solved upstream instead.

Not to sound rude, but I'd prefer if the software I install on my PC is official from Microsoft or the distro maintainers :/

**Comment by Levi--G** [comment]
Can we please put this back on the roadmap for .NET 10 since that will be the next LTS version. This is really hindering app developers from making .NET apps for for instance raspberry pi's since the installations would be too hard for the average consumer. Making us miss potential audiences for our software. this is **both hurting us developers and .NET adoption** for something that could be easily fixed! 

@NikolaMilosavljevic @MichaelSimons

**Comment by MichaelSimons** [comment]
@NikolaMilosavljevic, @jkoritzinsky - Does the infra added in https://github.com/dotnet/arcade/pull/15051 unblock the creation of arm64 packages?

**Comment by Aldaviva** [comment]
@AnalogFeelings No problem, that's completely understandable.

**Comment by richlander** [comment]
Would we do this for 8+?

**Comment by MichaelSimons** [comment]
> Would we do this for 8+?

The reason we didn't already have support arm64 packages was because the tooling we utilized didn't support arm64.  As part of the [Unified Build project](https://github.com/dotnet/arcade/blob/main/Documentation/UnifiedBuild/README.md), This infrastructure was rewritten to support a single linux vertical build (allowing all assets to be produced from a single leg to eliminate join points).  Additionally not all of the repos (e.g. sdk) utilized the shared arcade packaging infrastructure.  Backporting this work to 8.0 would be non-trivial and have risk.  We would need to make a business case to justify this work.

**Comment by Levi--G** [comment]
Ok so, 6 months later and:
-Documentation is still not in order
-Support is still not added

Can we please fix either for 2025?

I suggest either: 
Directly mentioning on the dotnet download page (https://dotnet.microsoft.com/en-us/download/dotnet/9.0) AMD64 only, and on the documentation page where it links (https://learn.microsoft.com/en-gb/dotnet/core/install/linux?WT.mc_id=dotnet-35129-website) the same warning in the right place.
Or:
Supporting other architectures that have binary builds

If you have no intention of doing the second please do the first one urgently. If i have users reporting bugs that "dotnet can't be installed" every dotnet project because they try to use package manager on a raspberry this becomes a real problem.

**Comment by richlander** [comment]
Here's a docs update to make the current situation clearer.

https://github.com/dotnet/docs/pull/44150

**Comment by AnalogFeelings** [comment]
So they're prioritizing support for Azure while other distributions are left in the dark with no ETA?

**Comment by richlander** [comment]
We're actively in the process of resolving this. Azure Linux went first because we did it all at once. Our Debian packaging workflow was x64 specific so had to be re-written. That's been the primary hold-up.

**Comment by AnalogFeelings** [comment]
Sorry if I'm being annoying, but hasn't the infrastructure been rewritten already?

**Comment by richlander** [comment]
I believe it has. I am hoping we will start publishing Arm64 packages this month or next.

**Comment by MichaelSimons** [comment]
We are targeting adding support in net10.0 (timeline is TBD).  We currently do not plan to backport this to previous version because of the expense of migrating the infrastructure.

**Comment by NikolaMilosavljevic** [comment]
@jkoritzinsky are additional PRs needed to enable arm64 Debs in `sdk` and `aspnetcore` repos?

**Comment by jkoritzinsky** [comment]
I think we may need additional PRs in those repos, yes.

**Comment by NikolaMilosavljevic** [comment]
Remaining work was completed with the following 3 PRs:
https://github.com/dotnet/arcade/pull/15564
https://github.com/dotnet/aspnetcore/pull/60565
https://github.com/dotnet/sdk/pull/47045
